pages/blog/event-driven-api-documentation-made-simple-clientside-rendering.md

```diff
@@ -0,0 +1,219 @@
+
+This guide is directed toward folks looking for instructions on how to generate documentation using their AsyncAPI files. Event-driven APIs are not the same as the synchronous APIs you usually document with OpenAPI or GraphQL. Many people use AsyncAPI now, and it is time to provide the community with a guide that shows what options the community has to render documentation on the client-side.
+
+Oh, if you are still not sure about Client-Side Rendering or Server-Side Rendering, then you can read all about it here 👉 [Website rendering for beginners](https://michals-corner.vercel.app/blog/website-rendering-for-beginners) (shameless plug 😉).
+
+I will cover the usage for:
+
+- [React](#react)
+- [Vue](#vue)
+- [Web Components](#wc)
+- [Standalone Bundle](#sb)
+
+All examples will use this same AsyncAPI sample file 👇
+
+```
+{
+  "asyncapi": "2.2.0",
+  "info": {
+    "title": "Account Service",
+    "version": "1.0.0",
+    "description": "This service is in charge of processing user signups"
+  },
+  "channels": {
+    "user/signedup": {
+      "subscribe": {
+        "message": {
+          "$ref": "#/components/messages/UserSignedUp"
+        }
+      }
+    }
+  },
+  "components": {
+    "messages": {
+      "UserSignedUp": {
+        "payload": {
+          "type": "object",
+          "properties": {
+            "displayName": {
+              "type": "string",
+              "description": "Name of the user"
+            },
+            "email": {
+              "type": "string",
+              "format": "email",
+              "description": "Email of the user"
+            }
+          }
+        }
+      }
+    }
+  }
+}
+```
+
+This is the expected look of the generated document 👇
+
+![document-template](/img/posts/event-driven-API-documentation-made-simple/document-template.webp "Document template")
+
+All usage examples from this article are available to check on [asyncapi-docs-rendering-examples](https://github.com/m1ner79/asyncapi-docs-rendering-examples) repository.
+
+---
+
+### <a name="react"></a>React
+
+If you wish to render documentation from your AsyncAPI file in React application, then you need to use [AsyncAPI React component](https://github.com/asyncapi/asyncapi-react).
+
+1️⃣ To install the **React AsyncAPI component** run the command `npm install --save @asyncapi/react-component@next`.
+
+2️⃣ Now, create **index.js** file and type in:
+
+```
+import React from "react";
+import ReactDOM from "react-dom";
+
+import AsyncApiComponent from "@asyncapi/react-component";
+import "@asyncapi/react-component/styles/default.min.css";
+
+import { specMock } from "./testDoc";
+
+const rootElement = document.getElementById("root");
+ReactDOM.render(<AsyncApiComponent schema={specMock} />, rootElement);
+```
+
+**React AsyncAPI component** is imported on the **line 4**.
+
+If you are happy with AsyncAPI styling then you need to import their CSS pattern with `import "@asyncapi/react-component/styles/default.min.css";`
```

Comment:
Such a code line should be transformed to the code block for better reading:

```suggestion
If you are happy with AsyncAPI styling then you need to import their CSS pattern with:

\```js
import "@asyncapi/react-component/styles/default.min.css";
\```
```

Please remove `/` before backticks - github markdown preview breaks markdown when you use multiline code blocks in another code blocks.